File: codes/python/custom_error_and_error_handling.py
print("\n# Custom Error and Error Handling in Python")


def give_me_respect(argument):
    if (argument != "respect"):
        # Custom Error
        raise Exception('Exception: You should give me "respect"!')
    return 'Thank you for giving me "respect"!'


# Error Handling
try:
    response = give_me_respect("boo!")
    print(response)
except Exception as any_exception:
    print(str(any_exception))
print("I'm sorry!")

# Without Error Handling
response = give_me_respect("boo!")
# this will never executed
print(response)
print("I'm sorry!")

File: codes/kotlin/custom_error_and_error_handling.kt
@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")

fun main() {
    println("\n// Custom Error and Error Handling in Kotlin")

    fun giveMeRespect(argument: String): String {
        // Custom Error
        if (argument != "respect") throw Exception("Exception: You should give me \"respect\"!")
        return "Thank you for giving me \"respect\"!"
    }

    var response: Any?

    // Error Handling
    try {
        response = giveMeRespect("boo!")
        println(response)
    } catch(anException: Exception) {
        println(anException.message)
    }
    println("I'm sorry!")

    // Without Error Handling
    response = giveMeRespect("boo!")
    // this will never executed
    println(response)
    println("I'm sorry!")
}
